packages/yew/src/lib.rs

```diff
@@ -295,7 +295,7 @@ mod renderer;
 
 #[cfg(feature = "csr")]
 #[cfg(test)]
-pub mod tests;
+pub(crate) mod tests;
```

Comment:
This needs to be `pub`, not `pub(crate)`. The tests module is part of the public API. Please just suppress the clippy lints here. 

See https://yew.rs/docs/more/testing#snapshot-testing

Reply:
I wasn't aware of the possibility for `#[cfg(test)]` to be transitive. AFAIK, it cannot be imported by dependents, `pub` or otherwise.

### `Cargo.toml`
```toml
[package]
name = "yewimport"
version = "0.1.0"
edition = "2021"

[dependencies]
yew = {version = "0.21.0", features = ["csr"]}
```
### `src/main.rs`
```rust
fn main() {
    
}

#[cfg(test)]
mod tests {
    use yew::tests::layout_tests::LayoutTest;
}
```
### `cargo test`
```console
$ cargo test
error[E0433]: failed to resolve: could not find `tests` in `yew`
   --> src/main.rs:7:14
    |
7   |     use yew::tests::layout_tests::LayoutTest;
    |              ^^^^^ could not find `tests` in `yew`
    |
note: found an item that was configured out
   --> /home/finnb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/yew-0.21.0/src/lib.rs:300:9
    |
300 | pub mod tests;
    |         ^^^^^
note: the item is gated here
   --> /home/finnb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/yew-0.21.0/src/lib.rs:299:1
    |
299 | #[cfg(test)]
    | ^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0433`.
error: could not compile `yewimport` (bin "yewimport" test) due to 1 previous error
```

But I can make the change you suggest!